### PR TITLE
Add user preferences to portal.

### LINKED ISSUE #1498: Bring back the slice list
The boxes with their auto-flow and such are difficult to scan with your eyes and find what you're looking for.  (Also they don't necessarily fit on the screen unless you have a huge monitor)

Having a single list where the columns are aligned with the data I'm looking for would be much better.

### LINKED ISSUE #1526: Add user preferences
Add a user preference capability. This could be used to support, for instance, #1482.

### PR DESCRIPTION
**Pull request by charliemeyer**
Retrieve user preferences with user-preferences.php, update them from the profile page via ajax using do-update-user-preferences.php. Store them in the database using the new preferences table located in update-12.sql. First supported preference (included in this pull request) is the ability to switch between the card view and a table view on the dashboard.

Fixes #1526.
Fixes #1498.

### CODE REVIEW

**Comment by charliemeyer** [comment]
Will update CHANGES when version gets bumped

**Comment by tcmitchell** [comment]
The new table should be added to the schema file in addition to the update file

File: lib/php/user-preferences.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    $sql = "SELECT * from user_preferences "
+    . "where user_urn = " . $conn->quote($user_urn, 'text')
+    . "and preference_name = " . $conn->quote($preference, 'text');
+    $db_res = db_fetch_row($sql);
+    $db_response = db_fetch_row($sql, "Get user preference");
+
+    $db_error = $db_response[RESPONSE_ARGUMENT::OUTPUT];
+    if($db_error != "") { // TODO: What do we do here
+      error_log("DB error when getting row from user_preferences table: " . $db_error);
+      return true;
```
Comment (tcmitchell) [inline lib/php/user-preferences.php]: return the default value instead: `$choices[0]`

File: lib/php/user-preferences.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+      if ($db_response[RESPONSE_ARGUMENT::VALUE]['preference_value']) {
+        return $db_response[RESPONSE_ARGUMENT::VALUE]['preference_value'];
+      } else {
+        $choices = $possible_prefs[$preference];
+        return $choices[0]; // the default value
+      }
+    }
+  } else {
+    error_log("Tried to load preference $preference that does not exist for user $user_urn");
```
Comment (tcmitchell) [inline lib/php/user-preferences.php]: Maybe "Unknown preference '$preference' requested for user '$user_urn'" would be clearer?

**Comment by tcmitchell** [comment]
Add `update-11.sql` to `portal/Makefile.am` so that it gets installed and can be loaded into the database.

**Comment by charliemeyer** [comment]
Above items corrected, CHANGES file all edited

**Comment by charliemeyer** [comment]
Fixed that bug with all the log messages about user_urn's

**Pull request merged**